bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -151,6 +163,62 @@ protected void processWatchEvent(WatchEvent<?> event, Kind<?> kind, Path path) {
             } catch (IOException e) {
                 logger.warn("Could not process config file '{}': {}", path, e);
             }
+        } else if (kind == ENTRY_DELETE) {
+            // Detect if a service specific configuration file was removed. We want to
+            // notify the service in this case with an updated empty configuration.
+            File configFile = path.toFile();
+            if (configFile.isHidden() || configFile.isDirectory() || !configFile.getName().endsWith(".cfg")) {
+                return;
+            }
+
+            exclusivcePIDMap.setFileRemoved(configFile.getAbsolutePath());
+            processOrphanExclusivePIDs();

```

Comment:
Shouldn't you also update the list on CREATE and MODIFIED events?

Reply:
absolutely. when the file is created during runtime and then removed when the system is shutdown the configuration will also remain in the configAdmin store.

Reply:
So why didn't you adapt the code then?